src/Collectors/JobCollector.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Collectors;
+
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\Collectors\Interfaces\DataCollectorInterface;
+use Illuminate\Foundation\Application;
+use Illuminate\Queue\Events\JobProcessed;
+use Illuminate\Queue\Events\JobProcessing;
+
+/**
+ * Collects info about the http request process
+ */
+class JobCollector extends TimelineDataCollector implements DataCollectorInterface
+{
+    protected $app;
+    protected $agent;
+
+    public function __construct(Application $app, Agent $agent, float $request_start_time)
+    {
+        $this->app = $app;
+        parent::__construct($request_start_time);
+
+        $this->agent = $agent;
+        $this->registerEventListeners();
+    }
+
+    public static function getName(): string
+    {
+        return 'job-collector';
+    }
+
+    protected function registerEventListeners(): void
+    {
+        $this->app->events->listen(JobProcessing::class, function () {
```

Comment:
I'm actually thinking that this should be opposite. The order of execution is JobProcessing, Middleware, JobProcessed. So as it is now the Job processing span extends a few milliseconds beyond the end of the transaction.

I did it this way to intentionally be similar to the HTTP Request, but since it's not fully accurate, I'm wondering if you'd prefer it switched--the transaction would be handled by the events and the middleware would just provide an additional span. This would also have the benefit of automatic transactions for pre-L6 installs.